sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -167,7 +170,12 @@ case class Alias(child: Expression, name: String)(
   override def metadata: Metadata = {
     explicitMetadata.getOrElse {
       child match {
-        case named: NamedExpression => named.metadata
+        case named: NamedExpression =>
+          new MetadataBuilder()
+            .withMetadata(named.metadata)
+            .remove(deniedMetadataKeys: _*)
```

Comment:
Do we really need a new public API for it? Can we just use a loop here?

Reply:
Fixed